waku/common/utils/timedmap.nim

```diff
@@ -0,0 +1,150 @@
+# Nim-LibP2P
+# Copyright (c) 2023 Status Research & Development GmbH
+# Licensed under either of
+#  * Apache License, version 2.0, ([LICENSE-APACHE](LICENSE-APACHE))
+#  * MIT license ([LICENSE-MIT](LICENSE-MIT))
+# at your option.
+# This file may not be copied, modified, or distributed except according to
+# those terms.
+
```

Comment:
For a new utility type it may be a good idea to preface the module with a brief description as to what is defined and what its purpose is (see for example https://github.com/status-im/nim-stew/blob/master/stew/keyed_queue.nim#L11-L23)